### PR TITLE
[2-17] Auditio module component

### PR DESCRIPTION
**Pull request by GeranSkot**
Added : 1. Audition component. 2. Audition module. 3. Audio-cloud.service (mock) 4. Audition service. Refactored code. Sorry for merging from that branch, approve plz :D

### CODE REVIEW

**Comment by KotdenGit** [comment]
All changes should be poured into the branch -dev.

**Comment by GeranSkot** [comment]
Sorry for this mess, I'll explain myself today

File: exalearn-ui/src/app/services/audition.service.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+			currentTime: undefined,
+			canplay: false,
+			error: false
+		};
+	}
+
+	getState(): Observable<StreamState> {
+		return this.stateChange.asObservable();
+	}
+}
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/audition.service.ts:130]: (this comment is about the whole file)
It seems that pausing audio wastes your attempt. Is it expected?
Can we at least warn user about it before the audio starts?

Maybe we should warn the user that the audio will start and disable/hide pause button?

File: exalearn-ui/src/app/services/utils.service.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+export function formatTimeAudio(time: number) {
+	return `${Math.floor(time / 60)} : ${String('0' + Math.floor(time % 60)).slice(-2)}`;
+}
+
+export function formatTimeTimer(minutes: number, seconds: number) {
+	const mins = minutes < 10 ? `0${minutes}` : String(minutes);
+	const secs = seconds < 10 ? `0${seconds}` : String(seconds);
+
+	return { mins, secs };
+}
```
Comment (vstrabko) [inline exalearn-ui/src/app/services/utils.service.ts:17]: Good job!

**Review by vstrabko — COMMENTED** [review]
Everything looks nice, but a couple of comments need to be addressed.
Did you check how audio module looks in mobile view?

**Review by vstrabko — APPROVED** [review]

File: exalearn-ui/src/app/components/audition/audition.component.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		<div class="player">
+			<mat-toolbar color="primary" class="player-container">
+				<mat-toolbar-row class="media-action-bar">
+					<button
+						mat-button
+						class="play-pause-button"
+						(click)="play(files[0], 0)"
+						[disabled]="state?.error"
+						*ngIf="!state?.playing"
+					>
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/audition/audition.component.html:31]: Please, move this lonely bracket

File: exalearn-ui/src/app/components/audition/audition.component.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+						[max]="state?.duration"
+						step="1"
+						[value]="state?.currentTime"
+						(input)="onSliderChangeEnd($event)"
+						[disabled]="state?.error || currentFile.index === undefined || state?.playing || !state?.playing"
+					></mat-slider>
+					{{ state?.readableDuration }}
+				</mat-toolbar-row>
+			</mat-toolbar>
+			<h1 class="attempts" *ngIf="NumberOfAttempts != 3">{{3 - NumberOfAttempts}} attempts left</h1>
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/audition/audition.component.html]: Maybe we can make it more flexible and configurable?

File: exalearn-ui/src/app/components/audition/audition.component.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+				[index]="i"
+				[currentIndex]="currentIndex"
+				(click)="currentIndex = i"
+			>
+			</app-test-nav-btn>
+			<button
+				mat-mini-fab
+				color="white"
+				class="nav-arrow-btn"
+				(click)="currentIndex < questions.length - 1 ? currentIndex = currentIndex + 1: currentIndex"
```
Comment (YanaBr) [inline exalearn-ui/src/app/components/audition/audition.component.html]: Please move it to separate file

**Review by YanaBr — CHANGES_REQUESTED** [review]
Please use whole grammar component inside your audition

**Review by HrebenYury — APPROVED** [review]

**Review by KotdenGit — APPROVED** [review]

**Review by YanaBr — APPROVED** [review]

**Pull request merged**